staging/src/k8s.io/apiserver/pkg/endpoints/handlers/update.go

```diff
@@ -184,6 +191,17 @@ func UpdateResource(r rest.Updater, scope *RequestScope, admit admission.Interfa
 			)
 			wasCreated = created
 			return obj, err
+		}
+		result, err := finishRequest(timeout, func() (runtime.Object, error) {
+			result, err := requestFunc()
+			if isTooLargeError(err) && scope.FieldManager != nil {
```

Comment:
This makes the code pretty twisty-turny, I don't have a better idea but maybe you can add a comment explaining why this is important? To all the places where we retry.